app/javascript/components/ApplicationsPairing.jsx

```diff
@@ -68,12 +75,70 @@ class ApplicationsPairing extends React.Component {
     )
   }
 
+  renderUpload(index) {
+    return (
+      <div key={index}>
+        <ActiveStorageProvider
+          endpoint={{
+            path: '/api/applications',
+            model: 'Application',
+            attribute: 'form',
+            method: 'POST',
+          }}
+          headers={{
+            'Content-Type': 'application/json'
+          }}
+          render={({ handleUpload, uploads, ready }) => (
```

Comment:
maybe separate this into a different function; traditionally we try to stay away from heavy inline function

Reply:
up to you though